x-pack/plugins/encrypted_saved_objects/server/crypto/encrypted_saved_objects_service.test.ts

```diff
@@ -2375,3 +2422,56 @@ describe('#decryptAttributesSync', () => {
     });
   });
 });
+
+describe('Embedding excluded AAD fields POC', () => {
+  it('decrypts using embedded AAD when available', async () => {
```

Comment:
Not sure if this is an adequate test or not. I simply created a new service and registered the encrypted type differently to simulate two versions. Are there other considerations we need to make as part of this POC?